.github/actions/setup-python-env/action.yml

```diff
@@ -1,14 +1,51 @@
-name: Setup Python env
-description: Install Python & Hatch
+name: Setup Python / Hatch / Pydantic Environment
+description: Install Python / Hatch / Pydantic
+
+inputs:
+  python-version:
+    description: "Version of Python to use for testing"
+    required: true
+  pydantic-version:
+    description: "Version of Pydantic to use for testing"
+    required: true
```

Comment:
If you set a default of 1.10 on this I think you can revert most of the non-caching updates in this PR, since I've separately consolidated actions into this setup.

Alternatively, we can use this as an annotation defaulted to "none" just to seed the cache key and then only install Pydantic in the calling action where it's required. That'll eliminate nearly all of the action-level changes in this PR. If we go that route we should probably name the key something like `installed-pydantic-version` or suchlike.